File: 2022/03/20/hammingDistance.ts
/**
 * https://leetcode.com/problems/hamming-distance/
 * @author Andrew Jarombek
 * @since 3/20/2022
 */

function hammingDistance(x: number, y: number): number {
    let count = 0;
    let z = x ^ y;

    while (z > 0) {
        if (z & 1)
            count++;

        z >>= 1;
    }

    return count;
}

File: 2023/06/28/go/hamming_distance.go
/**
 * https://leetcode.com/problems/hamming-distance/
 * Author: Andrew Jarombek
 * Date: 6/28/2023
 */

package _go

func hammingDistance(x int, y int) int {
	result := 0

	for x > 0 || y > 0 {
		if x%2 != y%2 {
			result++
		}

		x >>= 1
		y >>= 1
	}

	return result
}
